# Author: José Manuel Santamaría Lema <panfaust@gmail.com>
# Description: Fix for the following bug: the video widget i only displayed
# properly when it displays some subtitle, otherwise the widget displays trash.
# Forwarded: No, upstream have already fixed the issue.
--- subtitlecomposer-0.5.2.orig/src/widgets/textoverlaywidget.cpp
+++ subtitlecomposer-0.5.2/src/widgets/textoverlaywidget.cpp
@@ -64,6 +64,8 @@
 	m_textDocument->setTextWidth( width() );
 
 	updateColors();
+
+	paintEvent(0);
 }
 
 TextOverlayWidget::~TextOverlayWidget()
@@ -96,6 +98,7 @@
 {
 	if ( m_text != text )
 	{
+		setVisible( true );
 		m_text = text;
 		setDirty( true, false );
 	}
@@ -264,6 +267,8 @@
 
 void TextOverlayWidget::resizeEvent( QResizeEvent* e )
 {
+	setVisible( true );
+
 	if ( m_richTextMode )
 		m_textDocument->setTextWidth( e->size().width() );
 
@@ -272,6 +277,8 @@
 	m_bgPixmap = QPixmap( e->size() );
 
 	m_dirty = true;
+
+	paintEvent(0);
 }
 
 void TextOverlayWidget::updateColors()
@@ -300,7 +307,8 @@
 {
 	if ( m_text.isEmpty() )
 	{
-		setMask( m_noTextMask );
+		//setMask( m_noTextMask );
+		setVisible( false );
 	}
 	else
 	{
